internal/campaigns/types.go

```
@@ -284,6 +287,23 @@ func (c *Changeset) Clone() *Changeset {
 	return &tt
 }
 
+func (c *Changeset) SetMetadata(meta interface{}) {
+	switch pr := meta.(type) {
+	case *github.PullRequest:
+		c.Metadata = pr
+		c.ExternalID = strconv.FormatInt(pr.Number, 10)
+		c.ExternalServiceType = github.ServiceType
+		c.ExternalBranch = pr.HeadRefName
+		c.ExternalUpdatedAt = pr.UpdatedAt
+	case *bitbucketserver.PullRequest:
+		c.Metadata = pr
+		c.ExternalID = strconv.FormatInt(int64(pr.ID), 10)
+		c.ExternalServiceType = bitbucketserver.ServiceType
+		c.ExternalBranch = git.AbbreviateRef(pr.FromRef.ID)
+		c.ExternalUpdatedAt = unixMilliToTime(int64(pr.UpdatedDate))
+	}
```

Just a thought, not a hard requirement, but pointing it out in case you were on the fence about this: should we return an error if the metadata is of unknown type? Or should we log it?

---

I think let's return an error, it's what we do in other parts of the code